stdlib/array.ml

```diff
@@ -439,10 +439,18 @@ let stable_sort cmp a =
 
 let fast_sort = stable_sort
 
+let shuffle_contract_violation i j =
+  let msg =
+    "Array.shuffle: rand(" ^ string_of_int (i + 1) ^ ")" ^
+    " returned " ^ string_of_int j ^
+    ", outside of the expected range [0; " ^ string_of_int i ^ "]"
```

Comment:
note that using the more brief `out of expected range` makes it slightly more probable that the error message fits on a line (assuming 80 columns) and is thus easier to read.